R/clustering.R

```diff
@@ -676,79 +687,35 @@ RunLeiden <- function(
   weights = NULL,
   node.sizes = NULL,
   resolution.parameter = 1,
-  random.seed = 0
+  random.seed = 0,
+  n.iter = 10
 ) {
   if (!py_module_available(module = 'leidenalg')) {
     stop("Cannot find Leiden algorithm, please install through pip (e.g. pip install leidenalg).")
   }
-  # import python modules with reticulate
-  leidenalg <- import(module = "leidenalg")
-  ig <- import(module = "igraph")
-  # convert matrix input
-  adj_mat <- as.matrix(x = ceiling(x = adj_mat))
-  # convert to python numpy.ndarray, then a list
-  adj_mat_py <- r_to_py(x = adj_mat)
-  adj_mat_py <- adj_mat_py$tolist()
-  # convert graph structure to a Python compatible object
-  snn_graph <- ig$Graph$Adjacency(adj_mat_py)
-  # compute partitions
-  part <- switch(
-    EXPR = partition.type,
-    'RBConfigurationVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$RBConfigurationVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      resolution_parameter = resolution.parameter,
-      seed = as.integer(x = random.seed)
-    ),
-    'ModularityVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$ModularityVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      seed = as.integer(x = random.seed)
-    ),
-    'RBERVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$RBERVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      node_sizes = node.sizes,
-      resolution_parameter = resolution.parameter,
-      seed = as.integer(x = random.seed)
-    ),
-    'CPMVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$CPMVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      node_sizes = node.sizes,
-      resolution_parameter = resolution.parameter,
-      seed = as.integer(x = random.seed)
-    ),
-    'MutableVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$MutableVertexPartition,
-      initial_membership = initial.membership
-    ),
-    'SignificanceVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$SignificanceVertexPartition,
-      initial_membership = initial.membership,
-      node_sizes = node.sizes,
-      resolution_parameter = resolution.parameter
-    ),
-    'SurpriseVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$SurpriseVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      node_sizes = node.sizes
-    ),
-    stop("please specify a partition type as a string out of those documented")
+  if (method == "matrix") {
```

Comment:
Consider using `switch` instead of if/else statements. Swtiches are optimized for comparisons and are (mildly) faster

```R
> library(microbenchmark)
> cols <- c('red', 'blue', 'green')
> r1 <- microbenchmark({x <- sample(x = cols, size = 1); switch(EXPR = x, 'red' = 'yay', 'blue' = 'meh', 'nope'); })
> r2 <- microbenchmark({x <- sample(x = cols, size = 1); if (x == 'red') 'yay' else if (x == 'blue') 'meh' else 'nope'; })
> r1$expr <- 'switch'
> r2$expr <- 'if/else'
> rbind(r1, r2)
Unit: microseconds
    expr   min     lq    mean median     uq     max neval
 if/else 6.068 6.3380 7.29976 6.5735 6.8275  51.545   100
  switch 5.914 6.1365 9.27236 6.2505 6.5160 249.691   100
```

Reply:
Updated for performance (as has another if/switch statement added).